PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -3253,17 +3272,99 @@ struct HfTrackIndexSkimCreatorLfCascades {
           }
 
           // check not to take the same particle twice in the decay chain
-          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0DauPos.globalIndex() || trackPion1.globalIndex() == trackV0DauNeg.globalIndex() || trackPion1.globalIndex() == casc.globalIndex()) {
+          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0PosDau.globalIndex() || trackPion1.globalIndex() == trackV0NegDau.globalIndex()) {
             continue;
           }
 
           // primary pion track to be processed with DCAFitter
           auto trackParVarPion1 = getTrackParCov(trackPion1);
 
+          // find charm baryon decay using xi PID hypothesis
+          int nVtxFrom2ProngFitterXiHyp = df2.process(trackCascXi, trackParVarPion1);
+          if (nVtxFrom2ProngFitterXiHyp > 0) {
+
+            statusMassXiPiChannel = true;
+
+            df2.propagateTracksToVertex();
+
+            std::array<float, 3> pVecXi = {0.};
+            std::array<float, 3> pVecPion1XiHyp = {0.};
+            df2.getTrack(0).getPxPyPzGlo(pVecXi);
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1XiHyp);
+
+            auto secondaryVertex2XiHyp = df2.getPCACandidate();
+
+            if (std::abs(casc.mXi() - massXi) < cascadeMassWindow) {
+              SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi);
```

Comment:
This should be only set after you test `(mass2ProngXiHyp < massXiPiMin || mass2ProngXiHyp > massXiPiMax)` (line `3304` of current draft)

Reply:
Sure I can move it, but I think it would not make any difference because I fill the histograms and the table only if the charm baryon mass is within the specified range, so the bit would not appear anywhere if this condition is not satisfied

Reply:
Indeed, but it is misleading for those who read the code (also see my comment below, I suggest to remove the status that is redundant ;))